### DISCUSSION TITLE
[React] props와 state의 차이

### DISCUSSION THREAD

**Discussion by skyhanull** [wanted-9th-3team/frontend-study#5] [프론트 기술 질문 공간]
react에서 사용되는 props와 state의 차이는 무엇일까요?

**Comment by michoball** [comment]
react 컴포넌트에서 데이터를 다루는 개념에 props와 state가 있습니다. 

props는 react 컴포넌트 외부의 데이터를 컴포넌트 안에서 다루고자 할 때 사용되며 함수의 매게변수처럼 컴포넌트로 전달됩니다.  
 부모 컴포넌트에서 전달 된 데이터이기에 자식 컴포넌트에서는 값을 변경할 수 없습니다.  
 부모 컴포넌트에서 데이터가 변한다면 props로 전달 받은 자식 컴포넌트에서도 변한 데이터가 적용됩니다.

state는 react 컴포넌트 안에서 생성된 데이터로 함수 안에 선언한 변수처럼 컴포넌트 안에서 사용됩니다.
 컴포넌트가 마운트될 때 초기값으로 시작되며 컴포넌트 안에서 값을 변경할 수 있습니다. 
 
---

더 자세히는 작성하기 좀 힘드네요 ㅎㅎ  react 공식 문서랑 관련 글 링크 첨부해보겠습니다. 
https://ko.reactjs.org/docs/faq-state.html
https://lucybain.com/blog/2016/react-state-vs-pros/
https://github.com/uberVU/react-guide/blob/master/props-vs-state.md

**Comment by skyhanull** [comment]
props와 state는 react에서 데이터를 방법이라고 할 수 있습니다

state는 컴포넌트안에서 데이터를 변경할 수 있도록 하는 것입니다.
state는 배열의 형태로 state,setstate가 있으며 비동기적으로 작동합니다.

props는 부모컴포넌트에서 자식컴포넌트로 데이터를 전달하는 것을 말하며 읽기전용입니다

**Comment by WooGie911** [comment]
State는 가변 데이터로 구성요소에 의해 유지되며 변경이 가능하고 컴포넌트 내에서 관리됩니다.
반면 Props는 불변 데이터로 부모로부터 상속되며 변경이 불가합니다.

**Comment by BaekYuri** [comment]
props
- 부모 컴포넌트에서 자식 컴포넌트에게 값을 전달할 때 사용
- 읽기 전용으로 변경할 수 없음

state
- 현재 컴포넌트 내에서 생성하고, 변할 수 있는 데이터
- setState를 통해 변경할 수 있음

**Comment by bona373737** [comment]
state와 props의 대표적인 차이점은 데이터수정의 가능여부이다.

State는 내부 (컴포넌트)에서 생성하고 활동하고, 데이터를 변경할 수 있다. state값이 변경될때 컴포넌트는 재렌더링 된다. 
Props는 외부(부모 컴포넌트)에서 상속 받는 데이터이며, 데이터를 변경할 수 없다. 부모컴포넌트에서 값이 변경된 경우에 props를 전달받아 사용하는 자식컴포넌트도 함께 재렌더링 된다.
